File: ProgrammingIdeas/Scripts/OnSwipeListener.cs
using Android.App;
using Android.Content;
using Android.Views;
using Android.Widget;
using System;
using static Android.Views.GestureDetector;

namespace ProgrammingIdeas.Helpers
{
    internal enum Swipe
    {
        Left,
        Right
    }

    public class OnSwipeListener : Java.Lang.Object, View.IOnTouchListener
    {
        private static GestureDetector detector;

        public event EventHandler OnSwipeLeft;

        public event EventHandler OnSwipeRight;

        public OnSwipeListener(Context ctx)
        {
            detector = new GestureDetector(ctx, new GestureListener(SwipeRecieved));
        }

        private void SwipeRecieved(Swipe swipe)
        {
            switch (swipe)
            {
                case Swipe.Left:
                    OnSwipeLeft(this, new EventArgs());
                    break;

                case Swipe.Right:
                    OnSwipeRight(this, new EventArgs());
                    break;
            }
        }

        public bool OnTouch(View v, MotionEvent e)
        {
            return detector.OnTouchEvent(e);
        }

        private class GestureListener : SimpleOnGestureListener
        {
            private static int SWIPE_THRESHOLD = 150;
            private static int SWIPE_VELOCITY_THRESHOLD = 80;
            private Action<Swipe> baseListener;

            public GestureListener(Action<Swipe> listener)
            {
                baseListener = listener;
            }

            public override bool OnDown(MotionEvent e)
            {
                return true;
            }

            public override bool OnFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY)
            {
                bool result = false;
                try
                {
                    // Get vertical swipe distance.
                    float diffY = e2.GetY() - e1.GetY();

                    // Get horizontal swipe distance.
                    float diffX = e2.GetX() - e1.GetX();

                    // calculating swipe for x-axis.
                    if (Math.Abs(diffX) > Math.Abs(diffY) && (Math.Abs(diffX) > SWIPE_THRESHOLD && Math.Abs(velocityX) > SWIPE_VELOCITY_THRESHOLD))
                    {
                        if (diffX > 0)
                            SwipeRight();
                        else
                            SwipeLeft();
                        result = true;
                    }
                }
                catch
                {
                    Toast.MakeText(Application.Context, "Something went wrong.", ToastLength.Short).Show();
                }
                return result;
            }

            private void SwipeLeft()
            {
                baseListener(Swipe.Left);
            }

            private void SwipeRight()
            {
                baseListener(Swipe.Right);
            }
        }
    }
}

File: android/kotlin/app/src/main/java/com/alansa/ideabag2/utils/OnSwipeListener.kt
package com.alansa.ideabag2.utils

import android.content.Context
import android.view.GestureDetector
import android.view.MotionEvent
import android.view.View

enum class Swipe {
    LEFT,
    RIGHT
}

class OnSwipeListener(ctx: Context, val action: (Swipe) -> Unit) : View.OnTouchListener {
    private val detector = GestureDetector(ctx, GestureListener({ swipeReceived(it) }))
    override fun onTouch(p0: View?, p1: MotionEvent?): Boolean {
        return detector.onTouchEvent(p1)
    }

    fun swipeReceived(swipe: Swipe) {
        action(swipe)
    }

    class GestureListener(val action: (Swipe) -> Unit) : GestureDetector.SimpleOnGestureListener() {
        val SWIPE_THRESHOLD = 150
        val SWIPE_VELOCITY_THRESHOLD = 80

        override fun onDown(e: MotionEvent?): Boolean = true

        override fun onFling(e1: MotionEvent?, e2: MotionEvent?, velocityX: Float, velocityY: Float): Boolean {
            var result = false
            try {
                val diffY = e2!!.y - e1!!.y

                // Get horizontal swipe distance.
                val diffX = e2.x - e1.x

                // calculating swipe for x-axis.
                if (Math.abs(diffX) > Math.abs(diffY) && Math.abs(diffX) > SWIPE_THRESHOLD && Math.abs(velocityX) > SWIPE_VELOCITY_THRESHOLD) {
                    if (diffX > 0)
                        swipeRight()
                    else
                        swipeLeft()
                    result = true
                }
            } catch (e: Exception) {
            }

            return result
        }

        private fun swipeLeft() = action(Swipe.LEFT)


        private fun swipeRight() = action(Swipe.RIGHT)
    }
}